packages/rum-core/src/domain/rumEventsCollection/view/trackViews.ts

```diff
@@ -217,6 +204,10 @@ function newView(
     loadingType
   )
 
+  const { stop: stopReplayStatsTracking } = trackReplayStats(lifeCycle, id, (newReplayStats) => {
```

Comment:
It would be nice to be able to add replay metadata to views in view collection, something like:

```ts
function processViewUpdate(
  view: ViewEvent,
  foregroundContexts: ForegroundContexts,
  replayStats: any,
): RawRumEventCollectedData<RawRumViewEvent> {

const viewEvent = {
  ...
  ...replayStats.get()
}
...
}
```

Reply:
Okay, this was actually my first implementation. With this approach, we'd need to make sure the updated view is the one being recorded, which is not necessarily the case. My solution was to have a `getReplayStats(viewId)` function, to be sure to get the corresponding stats, but then we need to keep the stats indexed by view id just in case we need it again it the future, so it leaks memory.

Could we talk about it?

Reply:
Maybe instead of a stat object, you could use a replay observable. Would it solve the issue?